### DISCUSSION TITLE
log4net version

### DISCUSSION THREAD

**Discussion by BayMax619** [nhibernate/nhibernate-core#2933] [Q&A] [answered]
我当前使用NHibernate 1.2.0.2003, 此版本依赖了log4net 1.2.10.0
但是我想使用log4net 2.0.8.0
当我外部依赖log4net 2.0.8.0时, 报错找不到log4net 1.2.10.0.
我应该怎么做?

Google translate:
> I currently use NHibernate 1.2.0.2003, this version depends on log4net 1.2.10.0.
But I want to use log4net 2.0.8.0.
When I rely on log4net 2.0.8.0 externally, I get an error that log4net 1.2.10.0 cannot be found.
What should I do?

**Comment by hazzik** [comment, marked as answer]
Log4net 1.x and 2.x were signed by different public keys and so not compatible. So there are only 2 options for you:

1. Update NHibernate to version that does not have an explicit dependency on log4net (So the earliest version is 3.0.0.4000)
2. Compile NHibernate with replaced log4net version by yourself.
